lib/Swagger/src/TemplateClassification/Storage/Api/TCSApi.php

```diff
@@ -165,71 +169,65 @@ public function getTemplateDetailsWithHttpInfo($wiki_id, $page_id)
         $resourcePath = str_replace("{format}", "json", $resourcePath);
 
         
-        
-  
         // for model (json/xml)
         if (isset($_tempBody)) {
             $httpBody = $_tempBody; // $_tempBody is the method argument, if present
         } elseif (count($formParams) > 0) {
             $httpBody = $formParams; // for HTTP post (form)
         }
-        
         // make the API Call
         try {
             list($response, $statusCode, $httpHeader) = $this->apiClient->callApi(
-                $resourcePath, 'GET',
-                $queryParams, $httpBody,
-                $headerParams, '\Swagger\Client\TemplateClassification\Storage\Models\TemplateTypeProvider[]'
+                $resourcePath,
+                'GET',
+                $queryParams,
+                $httpBody,
+                $headerParams,
+                '\Swagger\Client\TemplateClassification\Storage\Models\TemplateTypeProvider[]',
+                '/{wiki_id}/{page_id}/providers'
             );
-            
-            if (!$response) {
-                return array(null, $statusCode, $httpHeader);
-            }
 
-            return array(\Swagger\Client\ObjectSerializer::deserialize($response, '\Swagger\Client\TemplateClassification\Storage\Models\TemplateTypeProvider[]', $httpHeader), $statusCode, $httpHeader);
-            
+            return [$this->apiClient->getSerializer()->deserialize($response, '\Swagger\Client\TemplateClassification\Storage\Models\TemplateTypeProvider[]', $httpHeader), $statusCode, $httpHeader];
         } catch (ApiException $e) {
-            switch ($e->getCode()) { 
-            case 200:
-                $data = \Swagger\Client\ObjectSerializer::deserialize($e->getResponseBody(), '\Swagger\Client\TemplateClassification\Storage\Models\TemplateTypeProvider[]', $e->getResponseHeaders());
-                $e->setResponseObject($data);
-                break;
+            switch ($e->getCode()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)